test/libsolidity/syntaxTests/operators/custom/operator_not_user_implementable.sol

```diff
@@ -0,0 +1,23 @@
+using {
+    f as <<,
+    f as >>,
+    f as **,
+    f as ++,
+    f as !,
+    f as x,
+    f as delete,
+    f as new,
+    f as ()
+} for int256;
+
+// ----
+// ParserError 4403: (17-19): The operator << cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
+// ParserError 4403: (30-32): The operator >> cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
+// ParserError 4403: (43-45): The operator ** cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
+// ParserError 4403: (56-58): The operator ++ cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
+// ParserError 4403: (69-70): The operator ! cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
+// ParserError 4403: (81-82): The operator cannot be user-implemented. This is only possible for the following operators: |, &, ^, +, -, *, /, %, ==, !=, <, >, <=, >=, ~
```

Comment:
Somehow I would also like to be able to use arbitrary function names as a parameter. E.g. `5 x 5`, or `5 times 5` - why not?

Reply:
Ah and the operator "name" is also not written here. Shouldn't it be `The operator x cannot be user-implemented` or `The operator 'x' cannot be user-implemented` .

Reply:
Maybe that's a restriction from the parser?

Reply:
It is. Operator name must be a valid token. And one that has a predefined value so identifiers are out.

Even if they weren't, that would clash with the literal suffixes that I'm working on top of this PR. You'll be able to define a function called `times` as a suffix and `5 times` would call that function. Then would it be `5 times 5` or `(5 times) 5`?

Another problem is that different operators have different signatures. For example `T == T` must return `bool` while `T + T` must return `T`. We wouldn't know which restriction to enforce on arbitrary operators.

Reply:
hmm.. just a random thought, that I just want to write down here.. lets say we allow arbitrary names for operators.. if we would allow unary postfix operators (not sure whether we have it right now, because I did not check yet) with arbitrary names, we could easily give the developer support to have arbitrary "units".

```solidity
type Meters is int128;

function _kilometers(Meters x) pure returns (Meters) {
     return Meters.wrap(Meters.unwrap(x) * 1000);
 }

function _meters(Meters x) pure returns (Meters) {
     return Meters.wrap(Meters.unwrap(x));
 }

contract C {
    type Meters is int128;
    using {
        _kilometers as km(postfix),
        _meters as m(postfix),
    } for Meters;

    function update(Meters m) public {
        //
    }

    function doSomething() public {
        update(5 km);
        update(3 m);
    }

} 
 ```